python/pyarrow/array.pxi

```diff
@@ -156,6 +156,14 @@ def array(object obj, type=None, mask=None, size=None, bint from_pandas=False,
 
         values = get_series_values(obj)
 
+        if np.ma.isMaskedArray(values):
+            if mask is not None:
+                raise ValueError("Cannot pass a numpy masked array and "
+                                 "specify a mask at the same time")
```

Comment:
I could also give preference to either the passed mask or the mask of the data, but to avoid confusion about which of the two is used, I decided to error for now. But can also update to any of the other options.

Reply:
I think avoid ambiguities is fine.